src/view/NicknameView.js

```diff
@@ -53,10 +53,13 @@ export default class NicknameView extends DatabaseView {
    * @inheritdoc
    */
   get isSelf () {
-    if (this.query.connection.state.user && this.object.user.id === this.query.connection.state.user.id) {
-      return this.query.connection.state.permissions.includes('nicknames.read.me')
+    if (!this.query.connection.state.user || !this.object.user) {
+      return false
     }
-    return false
+    if (this.object.user.id !== this.query.connection.state.user.id) {
```

Comment:
why not
```suggestion
    if (this.object.user?.id !== this.query.connection.state.user?.id) {
```

and then remove the null checks above.